sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtils.java

```diff
@@ -78,6 +77,10 @@ public static Optional<TextRange> toTextRange(InputFile inputFile, SonarAnalyzer
       endLineOffset = lineLength;
     }
 
+    if (startLine == endLine && startLineOffset == endLineOffset) {
```

Comment:
So this empty token can happen due to trimming?

Reply:
Yes exactly, trimming can make the range empty, which raises an exception and breaks tokenization.
I added a comment to make that clearer.